### PR TITLE
IGNITE-17698 Fix UUID serialization in configuration

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-17698

### CODE REVIEW

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java [hunk 1]
```diff
@@ -94,6 +97,16 @@ public static byte[] toBytes(Object value) {
                 return allocateBuffer(1 + strBytes.length).put(header).put(strBytes).array();
             }
 
+            case UUID: {
+                java.util.UUID uuid = (UUID) value;
```
Comment (ibessonov) [inline modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java:101]: What's up with fully qualified name?

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java [hunk 2]
```diff
@@ -208,6 +221,20 @@ public static byte[] toBytes(Object value) {
                 return buf.array();
             }
 
+            case UUID | ARRAY: {
+                java.util.UUID[] uuids = (UUID[]) value;
```
Comment (ibessonov) [inline modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java:225]: Same here

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java [hunk 3]
```diff
@@ -337,6 +367,16 @@ public static Serializable fromBytes(byte[] bytes) {
                 return res.toArray(EMPTY_STRING_ARRAY);
             }
 
+            case UUID | ARRAY: {
+                java.util.UUID[] uuids = new UUID[bytes.length / (Long.BYTES * 2)];
```
Comment (ibessonov) [inline modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java:371]: And here

**Review by ibessonov — CHANGES_REQUESTED** [review]

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java (hunk 1 above)
Comment (tkalkirill) [reply modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java]: Fix it

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java (hunk 2 above)
Comment (tkalkirill) [reply modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java]: Fix it

File: modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java (hunk 3 above)
Comment (tkalkirill) [reply modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java]: Fix it

**Review by ibessonov — APPROVED** [review]

**Pull request merged**